src/loader/AMASSLoader.py

```diff
@@ -0,0 +1,320 @@
+                    [key for key, value in AMASSLoader.supported_mocap_datasets.items()]))
+
+    @staticmethod
+    def _load_parametric_body_model(used_body_model_gender, num_betas, num_dmpls, data_path):
+        """
+        loads the parametric model SMPL+H based on the input parameters
+        :param used_body_model_gender: The model gender, pose will represented using male, female or neutral body shape. Type: string.
+        :param num_betas: Body shape parameters, used for BodyModel inistatiation
+        :param num_dmpls: deformation parameters, used for BodyModel inistantiation
+        :param data_path: path to the AMASS dataset root folder.
+        :return: parametric model instance BodyModel and face mesh values
+        """
+        bm_path = os.path.join(data_path, 'body_models/smplh',
+                               used_body_model_gender, 'model.npz')  # body model
+        dmpl_path = os.path.join(data_path, 'body_models/dmpls',
+                                 used_body_model_gender, 'model.npz')  # deformation model
+        if not os.path.exists(bm_path) or not os.path.exists(dmpl_path):
+            raise Exception(
+                "Parametric Body model doesn't exist, please follow download instructions section in AMASS Example")
+        comp_device = torch.device(
+            "cuda" if torch.cuda.is_available() else "cpu")
+        body_model = BodyModel(bm_path=bm_path, num_betas=num_betas, num_dmpls=num_dmpls, path_dmpl=dmpl_path).to(
+            comp_device)
+        faces = body_model.f.detach().cpu().numpy()
+        return body_model, faces
+
+    @staticmethod
+    def _get_supported_mocap_datasets(taxonomy_file_path, data_path):
+        """
+        get latest updated list from taxonomoy json file about the supported mocap datasets supported in the loader
+        module and update.supported_mocap_datasets list :param taxonomy_file_path: path to taxomomy.json file which
+        contains the supported datasets and their respective paths :param data_path: path to the AMASS dataset root
+        folder :return: None.
+        """
+        if os.path.exists(taxonomy_file_path):
+            with open(taxonomy_file_path, "r") as f:
+                loaded_data = json.load(f)
+                for block in loaded_data:
+                    if "sub_data_id" in block:
+                        sub_dataset_id = block["sub_data_id"]
+                        AMASSLoader.supported_mocap_datasets[sub_dataset_id] = os.path.join(
+                            data_path, block["path"])
+        else:
+            raise Exception(
+                "The taxonomy file could not be found: {}".format(taxonomy_file_path))
+
+    @staticmethod
+    def _clean_up_temp_dir(data_path):
+        """ Cleans up temporary generated poses directory """
+        temp_dir_path = os.path.join(data_path, "generated_poses")
+        if os.path.exists(temp_dir_path):
+            shutil.rmtree(temp_dir_path)
+
+    @staticmethod
```

Comment:
Why is this static if you then hand over the data_path which is an attribute of this class?

Reply:
for debugging reasons, I've drafted the module at first using only static functions. Now it is fixed.